### PR TITLE
feat: extract main generic_sender logic to send() python method

### PR DESCRIPTION
**Pull request by zlamalp**
- Previously all generic processing was part of the `__main__`, expected to be started from the cmd line only. Now it is extracted to the defined `send()` method and can be called directly from python scripts too.
- With this solution, we can still use the script from the command line, when `__main__` simply checks input and calls `send()` instead.
- New `opts` param was added to the `send()`, so the custom send scripts can extend default transport commands for SSH/cURL.

### CODE REVIEW

File: send/generic_sender.py [hunk 1]
```diff
@@ -95,18 +95,20 @@ def prepare_ssh_transport_command():
...
-	destination = sys.argv[2]
-	if len(sys.argv) == 3:
-		# if there is no destination type, use default 'host'
-		destination_type = send_lib.DESTINATION_TYPE_HOST
-	else:
-		destination_type = sys.argv[3]
-		if destination_type == send_lib.DESTINATION_TYPE_EMAIL or destination_type == send_lib.DESTINATION_TYPE_SERVICE_SPECIFIC:
-			print("Destination type " + destination_type + " is not supported yet.", file=sys.stderr)
-			exit(1)
+def main(facility_name: str, destination: str, destination_type: str = 'host', opts=None) -> None:
```
Comment (xflord) [inline send/generic_sender.py:98]:
```suggestion
def main(facility_name: str, destination: str, destination_type: str = send_lib.DESTINATION_TYPE_HOST, opts=None) -> None:
```
nitpick

**Review by xflord — APPROVED** [review]
I like this. Sad that I didn't think of it.

File: send/generic_sender.py (hunk 1 above)
Comment (zlamalp) [reply send/generic_sender.py]: Fixed, amended to previous commit.

**Review by HejdaJakub — APPROVED** [review]
It looks good from my side.

**Pull request merged**